spec/services/order_invoice_comparator_spec.rb

```diff
@@ -115,19 +115,241 @@
+            order.reload
+            expect(subject).to be true
+          end
+
+          it "phone" do
+            ship_address.update!(phone: "111-222-333")
+            order.reload
+            expect(subject).to be true
+          end
+
+          it "company" do
+            ship_address.update!(company: "A Company Name")
+            order.reload
+            expect(subject).to be true
+          end
+        end
+
+        context "customer changes on" do
+          let(:customer){ order.customer }
+
+          context "code" do
+            it "return false" do
+              customer.update!(code: 98_754)
+              order.reload
+              expect(subject).to be false
+            end
+          end
+
+          context "email" do
+            it "return false" do
+              customer.update!(email: "customer@email.org")
+              order.reload
+              expect(subject).to be false
+            end
+          end
+        end
+
+        context "payment changes on" do
+          let(:payment) { create(:payment, order_id: order.id) }
+          context "amount" do
+            it "returns true" do
+              payment.update!(amount: 222)
+              order.reload
+              expect(subject).to be true
+            end
+          end
+
+          context "payment method" do
+            let(:payment_method) { create(:payment_method) }
+
+            it "returns true" do
+              payment.update!(payment_method_id: payment_method.id)
+              order.reload
+              expect(subject).to be true
+            end
+          end
+
+          context "payment changes on" do
+            context "state" do
+              it "returns true" do
+                expect {
+                  payment.complete
+                }.to change { payment.state }.from("checkout").to("completed")
+
+                order.reload
+                expect(subject).to be true
+              end
+            end
+          end
+        end
       end
-    end
 
-    context "a relevant associated object is updated" do
-      let(:line_item){ order.line_items.first }
-      it "return true" do
-        line_item.update!(quantity: line_item.quantity + 1)
-        expect(subject).to be true
+      describe "ignoring non-relevant associated object changes" do
+        context "customer changes" do
+          let(:customer){ create(:customer) }
+          it "returns false" do
+            order.update!(customer_id: customer.id)
+            expect(subject).to be false
+          end
+        end
+
+        context "distributor changes" do
+          let(:distributor){ order.distributor }
+          it "returns false" do
+            distributor.update!(name: 'THIS IS A NEW NAME', abn: 'This is a new ABN')
+            expect(subject).to be false
+          end
+        end
+
+        context "order cycle changes" do
+          it "returns false" do
+            pending("WIP - can't seem to change the order cycle on an order")
```

Comment:
I'm getting a validation failure.
```ruby
     ActiveRecord::RecordInvalid:
       Validation failed: Distributor or order cycle cannot supply the products in your cart
```

Reply:
Discussed and removed on our live chat, resolving :pray: